<?xml version="1.0"?>
<launch>
    <arg name="map"         default=""/>
    <arg name="master_name" default="$(env MASTER)"/>
    <arg name="robot_name"  default="$(env HOST)"/>
    <arg     if="$(eval robot_name == '/')" name="frame_prefix" default=""/>
    <arg unless="$(eval robot_name == '/')" name="frame_prefix" default="$(arg robot_name)/"/>
    <arg name="map_frame"    default="$(arg frame_prefix)map"/>

    <include file="$(find xf_mic_asr_offline)/launch/mic_init.launch"/>

    <include file="$(find jetauto_navigation)/launch/navigation.launch">
        <arg name="map"         value="$(arg map)"/>
        <arg name="master_name" value="$(arg master_name)"/>
        <arg name="robot_name"  value="$(arg robot_name)"/>
    </include>

    <node pkg="xf_mic_asr_offline" name="voice_control_navigation" type="voice_control_navigation.py" output="screen" ns="$(arg master_name)">
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="costmap"   value="/$(arg frame_prefix)move_base/local_costmap/costmap"/>
    </node>

    <include file="$(find jetauto_navigation)/launch/rviz_navigation.launch"/>
</launch>
